### PR TITLE
IGNITE-21740 Fix zone definitions according to storage profiles feature

### PR DESCRIPTION
**Pull request by kgusakov**
https://issues.apache.org/jira/browse/IGNITE-21740

### CODE REVIEW

File: modules/api/src/main/java/org/apache/ignite/catalog/annotations/Zone.java [hunk 1]
```diff
@@ -52,9 +51,9 @@
...
 
     /**
-     * Engine name.
+     * Storage profiles.
      *
-     * @return Engine name.
+     * @return Storage profiles.
      */
-    ZoneEngine engine() default ZoneEngine.DEFAULT;
+    String storageProfiles();
```
Comment (alievmirza) [inline modules/api/src/main/java/org/apache/ignite/catalog/annotations/Zone.java:58]: Don't we need default here?

File: modules/api/src/main/java/org/apache/ignite/catalog/IgniteCatalog.java [hunk 2]
```diff
@@ -116,7 +116,7 @@
  *             .ifNotExists()
  *             .partitions(1)
  *             .replicas(3)
- *             .engine(ZoneEngine.AIMEM)
+ *             .storageProfiles("default")
```
Comment (alievmirza) [inline modules/api/src/main/java/org/apache/ignite/catalog/IgniteCatalog.java:119]: Please re-read the whole javadoc, seems that we didn't get rid of all mentions of `ENGINE`

File: modules/api/src/main/java/org/apache/ignite/catalog/annotations/Zone.java (hunk 1 above)
Comment (kgusakov) [reply modules/api/src/main/java/org/apache/ignite/catalog/annotations/Zone.java:58]: I think there is no appropariate default at this level of abstraction. storageProfiles is a mandatory field anyway, we have no defaults for it on any other levels at the moment.

File: modules/api/src/main/java/org/apache/ignite/catalog/IgniteCatalog.java (hunk 2 above)
Comment (kgusakov) [reply modules/api/src/main/java/org/apache/ignite/catalog/IgniteCatalog.java:119]: Fixed

**Review by alievmirza — APPROVED** [review]

**Pull request merged**